src/org/omegat/util/Language.java

```diff
@@ -84,20 +80,27 @@ public Language(Locale locale) {
      * This form is described in <a href="http://www.rfc-editor.org/rfc/bcp/bcp47.txt">BCP47</a>.
      */
     public Language(String str) {
-        this.languageCode = "";
-        this.countryCode = "";
-        this.locale = new Locale("");
         if (str != null) {
+            Locale checkLocale = Locale.forLanguageTag(str);
+
+            // Locale matches BCP 47
+            if (!checkLocale.getLanguage().isEmpty()) {
+                this.locale = checkLocale;
+                return;
```

Comment:
Instead of having multiple `return`s, this would be easier to follow if it was a simple `if ... else if ... else`.

Reply:
Good idea. This is fixed.